src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -480,8 +480,7 @@ private void removeAirNoLongerInTerritory() {
     // remove any air units that were once in this attack, but have now
     // moved out of the territory
     // this is an ilegant way to handle this bug
-    final CompositeMatch<Unit> airNotInTerritory = new CompositeMatchAnd<>();
-    airNotInTerritory.add(Matches.unitIsInTerritory(m_battleSite).invert());
+    final Match<Unit> airNotInTerritory = Matches.unitIsInTerritory(m_battleSite).invert();
```

Comment:
The method `MustFightBattle#removeAirNoLongerInTerritory()` was originally added in b3b6391:

```java
private void removeAirNoLongerInTerritory()
{
    //remove any air units that were once in this attack, but have now moved out of the territory
    //this is an ilegant way to handle this bug
    CompositeMatch airNotInTerritory = new CompositeMatchAnd();
    airNotInTerritory.add( new InverseMatch(Matches.unitIsInTerritory(m_territory)));
    airNotInTerritory.getMatches( m_attackingUnits, Matches.UnitIsAir);

    m_attackingUnits.removeAll(Match.getMatches(m_attackingUnits,airNotInTerritory));

}
```

and the line that includes `Matches#UnitIsAir` was removed in e6bb36b with a commit message of "eclipse inspired code cleanups."

At first, I thought this was a problem because I expected `Matches#UnitIsAir` to be involved in the composite match somehow.  However, after a double-take, I realized that the removed line wasn't being added to the composite match, but was a no-op and thus had no effect on the `m_attackingUnits.removeAll()` code below.

TL;DR: This code has been in the game since 2004.  It's either in the running for longest-lived bug ever, or the method was simply misnamed when it was introduced.